Add radio buttons & checkboxes

maya opened the pull request:
This adds radio buttons :radio: and checkboxes :heavy_check_mark: to the library. 

This addresses #210.

This is what it looks like:

![screen shot 2015-07-23 at 1 33 34 pm](https://cloud.githubusercontent.com/assets/5249443/8861441/32c05de4-313f-11e5-9a04-dd90f6aef2e4.png)

msecret commented on assets/_scss/elements/_inputs.scss:
```diff
@@ -57,4 +57,94 @@ label {
+    width: 1.8rem;
+  }
+
+  input[type="radio"] + span {
```
Will this or does this always have to be a `<span>`? Is there a chance that the user might want to make it another element besides a `<span>`?

maya replied:
Good point. @meiqimichelle wrote it this way so I wonder if using something else, like a class, would be more robust.

juliaelman replied:
+1 on making it a class and not locking into `<span>`

msecret commented on assets/_scss/elements/_inputs.scss:
```diff
@@ -57,4 +57,94 @@ label {
+    box-shadow: 0 0 0 1px $color-grey-600;
+    border-radius: $usa-border-radius;
+    display: inline-block;
+    height: 1.8rem;
```
Can you do 

```
margin {
  top:
  ...
}

maya replied:
Done in b41a36b.

fyi, it's 

```
margin: {
  top:
  ...
}
```

Need the colon after margin.

msecret commented on assets/_scss/elements/_inputs.scss:
```diff
@@ -57,4 +57,94 @@ label {
+
+label.radio :checked + span {
+  background-color: $color-primary;
+  border: 2px solid $color-white !important;
```
I'm not sure the `!important` rule is good here.

msecret commented on assets/_scss/elements/_inputs.scss:
```diff
@@ -57,4 +57,94 @@ label {
+  // to hide checkbox/radio from visual but 
+  // still read to screenreaders
+  border: 0;
+  clip: rect(0 0 0 0);
```
Will this work on all our supported browsers?

maya replied:
This will: 2844baa.

maya commented:
:+1: from @nickbristow on 508

arvalarva commented:
18f's writing style guide is using sentence case. You're using title case in this component. Can you explain why? We go back and forth with this issue A LOT here at the USPTO.

maya commented:
@arvalarva yes, we will be using sentence case. Fixed in 3da21a7. Will be updating throughout the site too. Updated the screenshot as well. Thanks for your feedback!

juliaelman commented on assets/_scss/elements/_inputs.scss:
```diff
@@ -57,4 +57,88 @@ label {
+  label {
+    margin-top: 0;
+  }
 }
```
New line at end of file.

juliaelman commented on assets/_scss/elements/_typography.scss:
```diff
@@ -116,7 +116,7 @@ dfn {
 
 .usa-sans,
-.usa-sans p {
+.usa-sans * {
```
Why are we using a wildcard here? Not sure this is best practice and would be hard for users to un-tangle.

juliaelman commented:
Closing this PR, as it looks as if #227 is an updated version to an updated fix.
